back/engines/commercial/analysis/app/policies/analysis/analysis_policy.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Analysis
+  class AnalysisPolicy < ApplicationPolicy
+    class Scope
+      attr_reader :user, :scope
+
+      def initialize(user, scope)
+        @user  = user
+        @scope = scope
+      end
+
+      def resolve
+        if user&.active? && user&.admin?
+          scope.all
+        elsif user&.active? && (user&.project_moderator? || user&.project_folder_moderator?)
+          projects = UserRoleService.new.moderatable_projects(user)
+          phases = Phase.where(project: projects)
+          scope.where(project: projects).or(scope.where(phase: phases))
+        else
+          scope.none
+        end
+      end
+    end
+
+    def index?
+      active? && (admin? || project_moderator? || project_folder_moderator?)
+    end
+
+    def create?
+      active? && (admin? || can_moderate?(object.hosting_project))
```

Comment:
The admin? check is not necessary when using the UserRoleService

Reply:
Makes sense. We have a `return true if user.admin?` guard clause as the first line in the user role service to do this optimization.